sdk/storage/azure_storage_blob/src/generated/models/pub_models.rs

```diff
@@ -256,7 +256,7 @@ pub struct BlobItemInternal {
 
     /// The name of the blob.
     #[serde(rename = "Name", skip_serializing_if = "Option::is_none")]
-    pub name: Option<BlobName>,
```

Comment:
At least this was able to get us out of one unnecessary layer (`BlobName` model is not necessary here), but this is still coming back as `Option<String>` when I expected pure `String`.

In TypeSpec we defined it as `name: string` and not `name?: string` in the `.tsp`. It doesn't really make sense in this context that a blob would have no name.